KNOWN_BUGS: Add entry 'Blocking socket operations'

jay opened the pull request:
- Add threaded resolver cleanup and GSSAPI for FTP to the TODO list of
  known blocking operations.

- New known bugs entry 'Blocking socket operations in non-blocking API'
  that directs to the TODO's list of known blocking operations.

Ref: https://github.com/curl/curl/pull/5214#issuecomment-612488021

Reported-by: Marc Hoersken

Closes #xxxx

bagder commented:
I think you could add those to `2.1 More non-blocking` in the TODO document instead, as that already lists some other places in the code that could have this kind of fix applied.

jay commented:
> I think you could add those to `2.1 More non-blocking` in the TODO document instead, as that already lists some other places in the code that could have this kind of fix applied.

That's where I added them, and then I added the KNOWN_BUGS entry to reference the TODO entry rather than duplicate the list.

bagder commented:
I wonder if we should perhaps rather merge those two documents into a single one...

jay commented:
There's some overlap but IMHO the TODO has a lot of things that are more enhancement requests than they are bugs.

bagder commented:
Yes, it is a spectrum but all the known bugs should be "todo issues" as well and if we feel we need to start adding pointers between the two document I think we've gone astray...

jay commented:
Aside from combining them (which I'm really meh about) do you have any objection to this in the meantime?

bagder commented:
Not at all, go ahead!
